src/game/Spell.cpp

```diff
@@ -5082,9 +5082,8 @@ SpellCastResult Spell::CheckCast(bool strict)
                 Creature* pet = m_caster->GetPet();
                 if (!pet)
                 {
-                    Pet* dbPet = new Pet;
-                    SpellCastResult result = dbPet->TryLoadFromDB((Player*)m_caster, 0);
-                    delete dbPet;
+                    Pet dbPet;
+                    SpellCastResult result = dbPet.TryLoadFromDB((Player*)m_caster, 0);
                     if(result != SPELL_FAILED_TARGETS_DEAD)
```

Comment:
This needs a space between if and (

also you maybe wanna rewrite it to:

`if (!dbPet.TryLoadFromDB((Player*)m_caster, 0))`